app/lib/common/widgets/html_editor/components/mention_handler.dart

```diff
@@ -0,0 +1,364 @@
+import 'package:acter/common/widgets/html_editor/components/mention_menu.dart';
+import 'package:acter/common/widgets/html_editor/models/mention_block_keys.dart';
+import 'package:acter/common/widgets/html_editor/models/mention_type.dart';
+import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+const double kMentionMenuHeight = 300;
+const double kMentionMenuWidth = 250;
+const double kItemHeight = 60;
+const double kContentHeight = 260;
+
+class MentionHandler extends ConsumerStatefulWidget {
+  const MentionHandler({
+    super.key,
+    required this.editorState,
+    required this.roomId,
+    required this.mentionType,
+    required this.onDismiss,
+    required this.onSelectionUpdate,
+    required this.style,
+  });
+
+  final EditorState editorState;
+  final String roomId;
+  final MentionType mentionType;
+  final VoidCallback onDismiss;
+  final VoidCallback onSelectionUpdate;
+  final MentionMenuStyle style;
+
+  @override
+  ConsumerState<MentionHandler> createState() => _MentionHandlerState();
+}
+
+class _MentionHandlerState extends ConsumerState<MentionHandler> {
+  final _focusNode = FocusNode();
+  final _scrollController = ScrollController();
+  List<MapEntry<String, String>> filteredItems = [];
+
+  int _selectedIndex = 0;
+  late int startOffset;
+  String _search = '';
+
+  @override
+  void initState() {
+    super.initState();
+    ref.listenManual(
+        mentionSuggestionsProvider((widget.roomId, widget.mentionType)),
+        (prev, next) {
+      if (next != null && next.isNotEmpty) {
+        filteredItems = next.entries.where((entry) {
+          final normalizedId = entry.key.toLowerCase();
+          final normalizedName = entry.value.toLowerCase();
+          final normalizedQuery = _search.toLowerCase();
+
+          return normalizedId.contains(normalizedQuery) ||
+              normalizedName.contains(normalizedQuery);
+        }).toList();
+      }
+    });
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      _focusNode.requestFocus();
+    });
+    startOffset = widget.editorState.selection?.endIndex ?? 0;
+  }
+
+  @override
+  void dispose() {
+    _scrollController.dispose();
+    _focusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Focus(
+      focusNode: _focusNode,
+      onKeyEvent: (node, event) => _handleKeyEvent(node, event, filteredItems),
```

Comment:
Why we are passing `filteredItems` as it is already global variable in class itself?